.dotnet/src/Custom/Chat/ChatTool.cs

```diff
@@ -3,80 +3,118 @@
     /// </summary>
+    /// <remarks> Present when <see cref="Kind"/> is <see cref="ChatToolKind.Function"/>. </remarks>
     public string FunctionDescription => Function?.Description;
 
-    // CUSTOM: Flattened.
+    // CUSTOM: Spread.
     /// <summary>
-    /// The parameter information for the function, provided in JSON Schema format.
+    ///     The parameters that the function accepts, which are described as a JSON schema. If omitted, this
+    ///     defines a function with an empty parameter list. Learn more in the
+    ///     <see href="https://platform.openai.com/docs/api-reference/chat/docs/guides/function-calling">function calling guide</see>
+    ///     and the
+    ///     <see href="https://json-schema.org/understanding-json-schema">JSON schema reference documentation</see>.
     /// </summary>
-    /// <remarks>
-    /// The <see cref="BinaryData.FromObjectAsJson{T}(T, System.Text.Json.JsonSerializerOptions?)"/> method provides
-    /// an easy definition interface using the <c>dynamic</c> type:
-    /// <para><code>
-    /// Parameters = BinaryData.FromObjectAsJson(new
-    /// {
-    ///     type = "object",
-    ///     properties = new
-    ///     {
-    ///         your_function_argument = new
-    ///         {
-    ///             type = "string",
-    ///             description = "the description of your function argument"
-    ///         }
-    ///     },
-    ///     required = new[] { "your_function_argument" }
-    /// })
-    /// </code></para>
-    /// </remarks>
+    /// <remarks> Present when <see cref="Kind"/> is <see cref="ChatToolKind.Function"/>. </remarks>
     public BinaryData FunctionParameters => Function?.Parameters;
 
-    public bool? StrictParameterSchemaEnabled => Function?.Strict;
-
-    // CUSTOM: Added custom constructor.
+    // CUSTOM: Spread.
     /// <summary>
-    /// Creates a new instance of <see cref="ChatTool"/>.
+    ///     <para>
+    ///         Whether to enable strict schema adherence when generating the function call. If set to <c>true</c>, the
+    ///         model will follow the exact schema defined in <see cref="FunctionParameters">
+    ///     </para>
+    ///     <para>
+    ///         Only a subset of the JSON schema specification is supported when this is set to <c>true</c>. Learn more
+    ///         about structured outputs in the
+    ///         <see href="https://platform.openai.com/docs/api-reference/chat/docs/guides/function-calling">function calling guide</see>.
+    ///     </para>
     /// </summary>
-    /// <param name="functionName"> The <c>name</c> of the function. </param>
-    /// <param name="functionDescription"> The <c>description</c> of the function. </param>
-    /// <param name="functionParameters"> The <c>parameters</c> into the function, in JSON Schema format. </param>
+    /// <remarks> Present when <see cref="Kind"/> is <see cref="ChatToolKind.Function"/>. </remarks>
+    public bool? StrictParameterSchemaEnabled => Function?.Strict;
```

Comment:
It would be nice if we could come up with a name that starts with "Function" for this property, so that all the properties from the "function" type appear together when sorted alphabetically in IntelliSense.

Reply:
Renamed to `FunctionSchemaIsStrict`